gcovr/gcov_parser.py

```diff
@@ -1024,16 +1051,67 @@ def _find_excluded_ranges(
                 else:  # pragma: no cover
                     pass
 
+            # branche exclusion
+
+            excl_branch_pattern = re.compile(
+                "(" + exclude_pattern_prefix + ")" + _EXCLUDE_BRANCH_PATTERN_POSTFIX
+            )
+            for header, flag in excl_branch_pattern.findall(code):
+
+                if flag == "LINE":
+                    if exclusion_stack_branch:
+                        warnings.branch_after_start(
+                            lineno, f"{header}_EXCL_LINE", exclusion_stack_branch[-1][1]
+                        )
+                    else:
+                        exclude_branch_ranges.append((lineno, lineno + 1))
+
+                if flag == "START":
+                    exclusion_stack_branch.append((header, lineno))
+
+                elif flag == "STOP":
+                    if not exclusion_stack_branch:
+                        warnings.stop_without_start(
+                            lineno, f"{header}_EXCL_START", f"{header}_EXCL_STOP"
+                        )
+                        continue
+
+                    start_header, start_lineno = exclusion_stack_branch.pop()
+                    if header != start_header:
+                        warnings.mismatched_start_stop(
+                            start_lineno,
+                            f"{start_header}_EXCL_START",
+                            lineno,
+                            f"{header}_EXCL_STOP",
+                        )
+
+                    exclude_branch_ranges.append((start_lineno, lineno))
+
+                else:  # pragma: no cover
+                    pass
+
         if exclude_lines_by_pattern_regex:
             if exclude_lines_by_pattern_regex.match(code):
                 exclude_line_ranges.append((lineno, lineno + 1))
```

Comment:
This should be moved to the end of the first loop.

Reply:
There is only one loop, that's why it's currently here.

We can split it into two loops, but that means going through the file twice. Is it worth it ?

Reply:
By mail I got a comment from you for this here but it isn't visible in the web interface. Did you remove it?

Reply:
Yes sorry, I should have only edit my comment. I was not looking at the correct line.

It's now part of the whole function `_process_exclusion_marker`

Reply:
No problem, I was only searching the comment and thought I'm old.